components/base/wheeled/wheeled_base.go

```diff
@@ -317,6 +317,42 @@ func (wb *wheeledBase) runAllGoFor(ctx context.Context, leftRPM, leftRotations,
 	return nil
 }
 
+// runAllSetRPM executes `motor.SetRPM` commands in parallel for left and right motors,
+// with specified speeds and rotations and stops the base if an error occurs.
+// All callers must register an operation via `wb.opMgr.New` to ensure the left and right motors
+// receive consistent instructions.
+func (wb *wheeledBase) runAllSetRPM(ctx context.Context, leftRPM, rightRPM float64) error {
+	if math.Abs(leftRPM) <= 10 || math.Abs(rightRPM) <= 10 {
+		wb.logger.CWarn(ctx, "low motor speed detected, motors may not behave as expected")
+	}
+	setRPMFuncs := func() []rdkutils.SimpleFunc {
+		ret := []rdkutils.SimpleFunc{}
+
+		// These reads of `wb.left` and `wb.right` can race with `Reconfigure`.
+		wb.mu.Lock()
+		defer wb.mu.Unlock()
+		for _, m := range wb.left {
+			motor := m
```

Comment:
This line is not needed. we can use m.SetRPM

Reply:
it actually is needed because without it if you have multiple motors in the `range wb.left`, it will repeatedly use the first motor. it was a weird bug that I found a while ago and I don't remember why this workaround works but it does. I will try to make a comment that explains it

Reply:
I wonder if what you saw is a symptom of something else. I'm not going to bar on this. But it seems like it's worth investigating why this happens.

Reply:
yeah it's definitely worth looking into again, gautham helped me debug originally so I'll ask him if he remembers what was up